cpp/src/Ice/WSEndpoint.cpp

```diff
@@ -214,64 +214,47 @@ IceInternal::WSEndpoint::transceiver() const
 }
 
 void
-IceInternal::WSEndpoint::connectors_async(EndpointSelectionType selType,
-                                         const EndpointI_connectorsPtr& callback) const
+IceInternal::WSEndpoint::connectorsAsync(
+    EndpointSelectionType selType,
+    function<void(const vector<IceInternal::ConnectorPtr>&)> response,
+    function<void(exception_ptr)> exception) const
 {
-    class CallbackI : public EndpointI_connectors
+    string host;
+    IPEndpointInfoPtr info = getIPEndpointInfo(_delegate->getInfo());
+    if(info)
     {
-    public:
-
-        CallbackI(const EndpointI_connectorsPtr& callback, const ProtocolInstancePtr& instance,
-                  const string& host, const string& resource) :
-            _callback(callback), _instance(instance), _host(host), _resource(resource)
-        {
-        }
+        ostringstream os;
+        os << info->host << ":" << info->port;
+        host = os.str();
+    }
 
-        virtual void connectors(const vector<ConnectorPtr>& c)
+    _delegate->connectorsAsync(
+        selType,
+        [response, host, this](const vector<ConnectorPtr>& connectors)
```

Comment:
Capturing "this" looks suspicious. You're sure it shouldn't be self = shared_from_this()?